src/librustc/traits/mod.rs

```diff
@@ -118,6 +118,7 @@ impl<'tcx> ObligationCause<'tcx> {
     }
 }
 
+/// The reason why a given obligation exists.
```

Comment:
Note that `ObligationCause` already has a doc comment, not sure if this is redundant.

Reply:
Had to go from `WfPredicates::cause` to `ObligationCauseCode` to `ObligationCodeCause` to find out that this is only relevant for diagnostics. I think that a short comment for `ObligationCauseCode` is useful, even if the this oneis kind of meh